apps/web/src/pages/templates/components/StepNameInput.tsx

```diff
@@ -0,0 +1,135 @@
+  const {
+    control,
+    formState: { errors, isSubmitted },
+  } = useFormContext<IForm>();
+  const { readonly } = useEnvController();
+  const showErrors = isSubmitted && errors?.steps;
+  const { colorScheme } = useMantineColorScheme();
+
+  return (
+    <Controller
+      control={control}
+      name={`steps.${index}.name`}
+      defaultValue={defaultValue}
+      render={({ field, fieldState }) => {
+        return (
+          <TextInput
+            styles={(theme) => ({
+              root: {
+                width: '85%',
+              },
+              wrapper: {
+                background: 'transparent',
+              },
+              input: {
+                background: 'transparent',
+                borderStyle: 'solid',
+                borderColor: colorScheme === 'dark' ? theme.colors.dark[5] : theme.colors.gray[5],
+                borderWidth: '1px',
+                fontSize: '20px',
+                fontWeight: 'bolder',
+                padding: 9,
+                lineHeight: '28px',
+                minHeight: 'auto',
+                height: 'auto',
+                width: '100%',
+                textOverflow: 'ellipsis',
+                '&:not(:placeholder-shown)': {
+                  borderStyle: 'none',
+                  padding: 10,
+                },
+                '&:hover, &:focus': {
+                  borderStyle: 'solid',
+                  padding: 9,
+                },
+              },
+            })}
+            {...field}
+            value={field.value || ''}
+            error={showErrors && fieldState.error?.message}
+            type="text"
+            data-test-id="step-name"
+            placeholder="Enter step name"
+            disabled={readonly}
+          />
+        );
+      }}
+    />
+  );
+};
+
+export const StepNameInput = ({
+  channel,
+  color = undefined,
+  index,
+}: {
+  channel: StepTypeEnum | ChannelTypeEnum;
+  index: number;
+  color?: any;
+}) => {
+  if (channel === StepTypeEnum.EMAIL || channel === ChannelTypeEnum.EMAIL) {
+    return (
+      <Group align="center" spacing={16}>
+        <Mail color={color} /> <Input defaultValue="Email" index={index} />
+      </Group>
+    );
+  }
+
+  if (channel === StepTypeEnum.IN_APP || channel === ChannelTypeEnum.IN_APP) {
+    return (
+      <Group align="center" spacing={16}>
+        <Bell color={color} /> <Input defaultValue="In-App" index={index} />
+      </Group>
+    );
+  }
+
+  if (channel === StepTypeEnum.CHAT || channel === ChannelTypeEnum.CHAT) {
+    return (
+      <Group align="center" spacing={16}>
+        <Chat color={color} /> <Input defaultValue="Chat" index={index} />
+      </Group>
+    );
+  }
+
+  if (channel === StepTypeEnum.PUSH || channel === ChannelTypeEnum.PUSH) {
+    return (
+      <Group align="center" spacing={16}>
+        <Mobile color={color} /> <Input defaultValue="Push" index={index} />
+      </Group>
+    );
+  }
```

Comment:
I agree with your comment but disagree with your solution I do not think it is good because channel is type and here we loose the types in the objects....